### PR TITLE
R4 Patient Read/Search/Write

### PR DESCRIPTION
**Pull request by ebaranowski**
Contains all the documentation to explain Patient Read, Search, and Write.

### CODE REVIEW

**Comment by MaxPhilips** [comment]
got this when nanoc-ing

```
kramdown warning(s) for <Nanoc::CompilationItemRepView item.identifier=/millennium/r4/individuals/patient.md name=default>
  Duplicate link ID '`date`' on line 1761 - overwriting      update  [0.01s]  output/millennium/r4/encounters/encounter/index.html

  No or invalid attributes found in IAL/ALD content: target=”_blank”
```

File: content/millennium/r4/individuals/patient.md [hunk 1]
```diff
@@ -9,4 +9,215 @@ title: Patient | R4 API
...
+ [`_count`]           | No                                           | [`number`] | The maximum number of results to return. Defaults to `20`.
+
+Notes:
+
+* Either the `_id`, or a combination of `identifier` , `birthdate`, `name`, `given`, `family`, `address-postalcode`, `phone`, or `email` parameters must be provided.
+* The `gender` parameter may only be provided if at least one of `identifier` , `birthdate`, `name`, `given`, `family`, `address-postalcode`, `phone`, or `email` parameters is provided.
+* The `name`, `family`, and `given` parameters support the ':exact' modifier and will search for current names only.
+* The `identifier`, `name`, `family`, `given`, `phone`, `email`, `address-postalcode`, or `gender` parameters may be provided exactly once and may have only a single value.
+* The `birthdate` parameter may be provided twice to indicate a date range, but must contain the inclusive prefixes 'le' and 'ge'
+* The `birthdate` parameter may be provided once with the following prefixes: 'ge', 'le', 'gt', 'lt', 'eq'
```
Comment (MaxPhilips) [inline content/millennium/r4/individuals/patient.md]: can we change this to

```
* The `birthdate` parameter
  * May be provided twice using the prefixes 'le' and 'ge' to indicate a date range
  * May be provided once using one of the following prefixes to imply a date range: 'ge', 'le', 'gt', 'lt', 'eq'
```

do we allow dates with times?

File: content/millennium/r4/individuals/patient.md [hunk 2]
```diff
@@ -9,4 +9,215 @@ title: Patient | R4 API
...
+
+### Headers
+
+ <%= headers %>
```
Comment (MaxPhilips) [inline content/millennium/r4/individuals/patient.md]: rebase when available to pull in this change for fhir+json https://github.com/cerner/fhir.cerner.com/commit/761f781e1eb58169b9b02d2530719b468b68a24d#diff-7718c11ed130ac00a9854e0b089f74b4L27

File: content/millennium/r4/individuals/patient.md [hunk 3]
```diff
@@ -9,4 +9,215 @@ title: Patient | R4 API
...
+
+    POST https://fhir-ehr.sandboxcerner.com/r4/0b8a0111-e8e6-4c26-a91c-5069cbc6b1ca/Patient
+
+#### Body
+
+#### Response
+
+<%= headers status: 201 %>
+<pre class="terminal">
+    Cache-Control: no-cache
```
Comment (MaxPhilips) [inline content/millennium/r4/individuals/patient.md]: spaces inside a pre tag are rendered as spaces on the site

File: lib/resources/example_json/r4_example_patient.rb [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+      "id": "4342009",
+      "meta": {
+        "versionId": "43",
+        "lastUpdated": "2019-03-27T15:51:22.000Z"
+      },
+      "text": {
+        "status": "generated",
+        "div": "&lt;div&gt;&lt;p&gt;&lt;b&gt;Patient&lt;/b&gt;&lt;/p&gt;&lt;p&gt;&lt;b&gt;Status&lt;/b&gt;: Active&lt;/p&gt;&lt;p&gt;&lt;b&gt;Name&lt;/b&gt;: SMART, NANCY&lt;/p&gt;&lt;p&gt;&lt;b&gt;DOB&lt;/b&gt;: Aug 11, 1980&lt;/p&gt;&lt;p&gt;&lt;b&gt;Administrative Gender&lt;/b&gt;: Female&lt;/p&gt;&lt;p&gt;&lt;b&gt;Marital Status&lt;/b&gt;: Married&lt;/p&gt;&lt;/div&gt;"
+      },
+      "extension": [{
```
Comment (MaxPhilips) [inline lib/resources/example_json/r4_example_patient.rb]: the site renders this appropriately but I would prefer if we send this through a json formatter here as well

File: lib/resources/example_json/r4_example_patient.rb [hunk 5]
```diff
@@ -0,0 +1,338 @@
+module Cerner
```
Comment (MaxPhilips) [inline lib/resources/example_json/r4_example_patient.rb]: move all json to r4_examples_patient.rb

File: lib/resources/r4/patient.yaml [hunk 6]
```diff
@@ -0,0 +1,366 @@
...
+          "system": "urn:oid:1.1.1.1.1.1",
+          "value": "&lt;unique identifier&gt;",
+          "period": {
+            "start": "2016-01-02T00:00:00-05:00",
+            "end": "2020-01-02T00:00:00-05:00"
+          }
+        }
+      ]
+    }
+  note: Atleast one identifier should be provided, and must only contain an `assigner` field with a reference to the Organization in which the patient is being enrolled.
```
Comment (MaxPhilips) [inline lib/resources/r4/patient.yaml]: let's use bullets to break this down, html ul and li tags should work here

same for name

File: lib/resources/example_json/r4_example_patient.rb (hunk 5 above)
Comment (MaxPhilips) [reply lib/resources/example_json/r4_example_patient.rb]: looks like Zameer got this in the patch PR

File: content/millennium/r4/individuals/patient.md (hunk 1 above)
Comment (ebaranowski) [reply content/millennium/r4/individuals/patient.md]: No, we do not allow dates with times.

Comment (MaxPhilips) [reply content/millennium/r4/individuals/patient.md]: then let's call that out too

File: lib/resources/example_json/r4_example_patient.rb (hunk 4 above)
Comment (ebaranowski) [reply lib/resources/example_json/r4_example_patient.rb]: Screen Shot 2019-03-28 at 10 25 53 AM

File: content/millennium/r4/individuals/patient.md (hunk 3 above)
Comment (MadhurRaj) [reply content/millennium/r4/individuals/patient.md]: Yes, and we've used 4 spaces while rendering them for DSTU2. Are we changing this for R4?

Comment (MaxPhilips) [reply content/millennium/r4/individuals/patient.md]: yes

File: lib/resources/example_json/r4_example_patient.rb (hunk 4 above)
Comment (MaxPhilips) [reply lib/resources/example_json/r4_example_patient.rb]: right, the site renders it regardless of what spacing we have here. I would prefer the raw json to match the formatted json

**Comment by ebaranowski** [comment]
> got this when nanoc-ing
> 
> ```
> kramdown warning(s) for <Nanoc::CompilationItemRepView item.identifier=/millennium/r4/individuals/patient.md name=default>
>   Duplicate link ID '`date`' on line 1761 - overwriting      update  [0.01s]  output/millennium/r4/encounters/encounter/index.html
> 
>   No or invalid attributes found in IAL/ALD content: target=”_blank”
> ```

Fixed the rendering issues: https://github.com/cerner/fhir.cerner.com/pull/241/commits/8b4a41befb353cad101bd7e61edcbdcce61e32d5#diff-fb8aa00e1932d86c4425f2f33f20c7d1R214

File: content/millennium/r4/individuals/patient.md (hunk 1 above)
Comment (ebaranowski) [reply content/millennium/r4/individuals/patient.md]: Updated: https://github.com/cerner/fhir.cerner.com/pull/241/commits/8b4a41befb353cad101bd7e61edcbdcce61e32d5#diff-fb8aa00e1932d86c4425f2f33f20c7d1R79

File: content/millennium/r4/individuals/patient.md (hunk 3 above)
Comment (ebaranowski) [reply content/millennium/r4/individuals/patient.md]: Removed spacing: https://github.com/cerner/fhir.cerner.com/pull/241/commits/8b4a41befb353cad101bd7e61edcbdcce61e32d5#diff-fb8aa00e1932d86c4425f2f33f20c7d1L191

File: content/millennium/r4/individuals/patient.md (hunk 2 above)
Comment (ebaranowski) [reply content/millennium/r4/individuals/patient.md]: Rebased this branch

File: lib/resources/example_json/r4_example_patient.rb (hunk 4 above)
Comment (ebaranowski) [reply lib/resources/example_json/r4_example_patient.rb]: Adjusted the brackets: https://github.com/cerner/fhir.cerner.com/pull/241/commits/830d1fd3a082e2e565f0cd28b38c79997d97c597

Comment (MaxPhilips) [reply lib/resources/example_json/r4_example_patient.rb]: 907614b

Comment (ebaranowski) [reply lib/resources/example_json/r4_example_patient.rb]: https://github.com/cerner/fhir.cerner.com/pull/241/commits/907614b11e96ebb5bd585832252c24e1b8c16d8d

File: lib/resources/r4/patient.yaml (hunk 6 above)
Comment (MadhurRaj) [reply lib/resources/r4/patient.yaml]: updated: 0bcdebffd7683a06e5014f464bf3715d627d7b63

**Comment by MadhurRaj** [comment]
rebased with master.

**Comment by MadhurRaj** [comment]
rebased to pull in the types links

**Pull request merged**